src/App.tsx

```diff
@@ -1,93 +1,136 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { TodosFilter } from './components/TodosFilter';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { SortBy, Todo } from './types';
+
+let todos = localStorage.getItem('todosStorage');
+let todosFromLocalStorage: Todo[] = [];
+
+if (todos !== null) {
+  todosFromLocalStorage = JSON.parse(todos);
+  if (!Array.isArray(todosFromLocalStorage)) {
+    todosFromLocalStorage = [];
+  }
+}
 
 export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+  const [completedTodosId, setCompletedTodosId] = useState<number[]>(
+    todosFromLocalStorage.filter(todo => todo.completed)
+      .map(todoId => todoId.id),
+  );
```

Comment:
no needs to move this value to the state. It's also a calculated value, so you can just use useMemo

Reply:
completedTodosId it is a array of completed todos id numbers only. I use and change this in many function so is easier to use it as state.